obj.js

```diff
@@ -128,13 +158,21 @@ export default class BaseProtocol {
     let stub = Object.create(proto);
     stub.id = objDescrJSON.idLocal;
     this.addRemoteObject(objDescrJSON.idLocal, stub);
-    for (let propName in objDescrJSON.properties) {
-      stub[propName] = await this.mapIncomingObjects(objDescrJSON.properties[propName]);
-    }
+    await this.updateObjectProperties(stub, objDescrJSON.properties);
 
     return stub;
   }
 
+  async updateObjectProperties(obj, properties) {
+    for (let propName in properties) {
+      Object.defineProperty(obj, propName, {
+        configurable: true,
+        enumerable: true,
+        value: await this.mapIncomingObjects(properties[propName]),
+      });
```

Comment:
What's the difference to this? (You probably have a reason, but I don't see it.)
```suggestion
obj[propName] = await this.mapIncomingObjects(properties[propName]);
```

 Could you please add a code comment why you use `defineProperty()` here?

Reply:
I don't remember for sure; I may have been defensive, not wanting to accidentally trigger a setter by mistake, but I might also have wanted to make it clear that these values are meant to be read only (at least where the regular assignment operator is concerned).

Reply:
OK, I see. An explicit `writable: false,` (despite it being the [default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty)) would help readability, then.